### PR TITLE
CBP 22: Added EmailLetter class, added Celery task for sending emails

### PR DESCRIPTION
**Pull request by pazuzu-ua**

### CODE REVIEW

File: backend/apps/emails/models.py [hunk 1]
```diff
@@ -137,26 +146,134 @@ def _generate_keywords(self, candidate: Candidate = None, vacancy: Vacancy = Non
...
+        to=Vacancy,
+        verbose_name=_("Associated Vacancy"),
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+    )
+    recipients = models.ManyToManyField(
+        to=Candidate,
+        verbose_name=_("Recipients"),
+    )
```
Comment (devoorhees) [inline backend/apps/emails/models.py]: Please, add `related_name` attributes to these fields.

File: backend/apps/emails/models.py [hunk 2]
```diff
@@ -137,26 +146,134 @@ def _generate_keywords(self, candidate: Candidate = None, vacancy: Vacancy = Non
...
+
+    @property
+    def sent_time(self) -> str:
+        if self.sent_at:
+            return self.sent_at
+        return "Not available"
```
Comment (devoorhees) [inline backend/apps/emails/models.py]:
```
def sent_time(self) -> datetime.datetime | str:
    return self.sent_at or "Not available"
```

File: backend/apps/emails/tasks.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                    vacancy=email_letter.vacancy,
+                    event=email_letter.event,
+                ),
+                html_message=template.render_body(
+                    context=context,
+                    candidate=candidate,
+                    vacancy=email_letter.vacancy,
+                    event=email_letter.event,
+                ),
+                message=None,
```
Comment (devoorhees) [inline backend/apps/emails/tasks.py]: You can add:
```
from django.utils.html import strip_tags
message=strip_tags(html_message)
```
I guess, `message` is used if it's not able to render `html_message` so we still have some meaningful text inside the sending email.

**Review by devoorhees — APPROVED** [review]

File: backend/apps/emails/tasks.py (hunk 3 above)
Comment (pazuzu-ua) [reply backend/apps/emails/tasks.py]: That will attach them both. It uses EmailMultiAlternatives under the hood and if it succeeds, we will dublicate the message then...

Comment (devoorhees) [reply backend/apps/emails/tasks.py]: It works and doesn't duplicate via gmail backend. Checked.

**Review by LV-coding — APPROVED** [review]

**Pull request merged**